File: SysWas.Domain/Entidades/Financeiro/Custo.cs
using System;
using System.Collections.Generic;
using System.Text;
using SysWas.Domain.Entidades.Utilitarios;


namespace SysWas.Domain.Entidades.Financeiro
{
    public class Custo : Entidade
    {
        public int Id { get; set; }
        public double Valor { get; set; }

        public DateTime DataEmissao { get; set; }
        public DateTime DataVencimento { get; set; }
        public bool Pago { get; set; }


        public int TipoId { get; set; }
        public virtual TipoCusto Tipo { get; set; }
        public int PagamentoId { get; set; }
        public virtual FormaPagamento Pagamento { get; set; }

        public int AprovavaoId { get; set; }
        public virtual EstadoAprovacao Aprovacao { get; set; }


        public override void Validate()
        {
            //if (this.Id == 0)
            //    AddMessageValidation("Não foi identificado qual a referencia desse item!");
        }


    }
}

File: SysWas.Web/ClientApp/src/app/model/financeiro/custo.ts
import { TipoCusto } from './tipoCusto';
import { FormaPagamento } from '../utilitarios/formaPagamento';
import { EstadoAprovacao } from '../utilitarios/estadoAprovacao';

export class Custo {

    id: number;
    valor: number;

    dataEmissao: Date;
    dataVencimento: Date;

    pago: boolean;

  tipoId: number;
  tipo: TipoCusto;
  pagamentoId: number;
  pagamento: FormaPagamento;
  aprovacaoId: number;
  aprovacao: EstadoAprovacao;

}
